### ISSUE TITLE
as.undirected() has problems with multiple edges

### ISSUE THREAD

**Issue by gaborcsardi** [igraph/rigraph#35]
_From @g-chi on August 1, 2014 3:5_

The graph is
a b 1
a b 2
b a 3
a a 4
a a 2

g = read.graph('this_file.ncol', format = 'ncol', directed = TRUE)
when I use the as.undirected(g,mode="collapse"), I get
a a 7
a b 5

What I expect is
a a 6
a b 6

So as.undirected() has problems with multiple edges.
http://stackoverflow.com/questions/25030345/what-is-the-mode-collapse-in-as-undirected-function

_Copied from original issue: igraph/igraph#667_

**Comment by ntamas** [comment]
Seems to be a bug in the C core as the same result is observed from the Python interface. The bug is probably triggered by the presence of loop edges because the result is correct if I remove the loop edges first.

**Comment by ntamas** [comment]
https://github.com/igraph/igraph/issues/1814

**Comment by ntamas** [comment]
This is now fixed in the upcoming version 0.9.5 of the C core. Ultimately our goal is to release version 1.3.0 of the R interface with a full upgrade to igraph's most recent 0.9.x C core so this will most likely be fixed when R-igraph 1.3.0 is released.

**Comment by szhorvat** [comment]
Quick test code:

```
g<-graph_from_literal(a-+b, a-+b, b-+a, a-+a, a-+a, simplify = F)
E(g)$weight <- c(1,2,3,4,2)
as_data_frame(as.undirected(g, mode='collapse'))
```

Works in 1.3 dev, closing.
